app/actions/websocket/index.test.ts

```diff
@@ -0,0 +1,215 @@
+import EphemeralStore from '@store/ephemeral_store';
+import NavigationStore from '@store/navigation_store';
+
+import {handleFirstConnect, handleReconnect} from './index';
+
+import type PostModel from '@typings/database/models/servers/post';
+import type UserModel from '@typings/database/models/servers/user';
+
+jest.mock('@actions/local/channel');
+jest.mock('@actions/local/systems');
+jest.mock('@actions/remote/channel');
+jest.mock('@actions/remote/entry/common');
+jest.mock('@actions/remote/post');
+jest.mock('@actions/remote/preference');
+jest.mock('@actions/remote/user');
+jest.mock('@calls/actions/calls');
+jest.mock('@calls/utils');
+jest.mock('@database/manager');
+jest.mock('@managers/apps_manager');
+jest.mock('@queries/app/servers');
+jest.mock('@queries/servers/post');
+jest.mock('@queries/servers/system');
+jest.mock('@queries/servers/thread');
+jest.mock('@queries/servers/user');
+jest.mock('@store/ephemeral_store');
+jest.mock('@store/navigation_store');
+jest.mock('@store/team_load_store');
+jest.mock('@utils/helpers', () => ({
+    isTablet: jest.fn().mockReturnValue(false),
+}));
+
+describe('WebSocket Index Actions', () => {
+    const serverUrl = 'baseHandler.test.com';
+    const currentUserId = 'current-user-id';
+    const currentTeamId = 'current-team-id';
+    const currentChannelId = 'current-channel-id';
+    const groupLabel = 'DeepLink';
+
+    beforeEach(async () => {
+        jest.clearAllMocks();
+        jest.spyOn(DeviceEventEmitter, 'emit');
+        await DatabaseManager.init([serverUrl]);
+        DatabaseManager.serverDatabases[serverUrl] = {
+            operator: {
+                database: {},
+                batchRecords: jest.fn(),
+            },
+        } as any;
+        DatabaseManager.getServerDatabaseAndOperator = jest.fn().mockReturnValue({
+            database: {},
+            operator: {
+                batchRecords: jest.fn(),
+            },
+        });
+    });
+
+    describe('handleFirstConnect', () => {
+        it('should handle first connection successfully', async () => {
+            const mockEntryData = {
+                models: [],
+                initialTeamId: currentTeamId,
+                initialChannelId: currentChannelId,
+                prefData: {preferences: []},
+                teamData: {memberships: [], teams: []},
+                chData: {memberships: [], channels: []},
+                gmConverted: false,
+            };
+
+            jest.mocked(entry).mockResolvedValue(mockEntryData);
+            jest.mocked(getCurrentUser).mockResolvedValue({
+                id: currentUserId,
+                locale: 'en',
+            } as UserModel);
+            jest.mocked(getConfig).mockResolvedValue({Version: '9.0.0'} as ClientConfig);
+            jest.mocked(isSupportedServerCalls).mockReturnValue(true);
+
+            const result = await handleFirstConnect(serverUrl, groupLabel);
+
+            expect(result).toBeUndefined();
```

Comment:
we should name it error, instead of result otherwise is confusing that the expected result is undefined.